### ISSUE TITLE
PyTorch model precision

### ISSUE THREAD

**Issue by dereyly** [jwyang/faster-rcnn.pytorch#60]
I try to reproduce preciosn with pytorch origin resnet101
And have only Mean AP = 0.7217
Why precision gap is so large?

My experiments
1. Start origin training and get mAp=0.748 with bs=4 and lr=4e-3
2. Start training with pretrained resnet101 and ignore caffe converted moedl
and have loss= nan
2.1. Train with clip_gardients mAp=0.64
3. Train with classic pytorch normalization ( to avoid problem in config set  pixel_mean=[0,0,0])
get mAp=0.7217

**Comment by jwyang** [comment]
@dereyly you should preprocess the images as the pytorch using the normalization way. please read the read me.

**Comment by dereyly** [comment]
@jwyang Thank you for response, I use standard pytorch normalization
normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],
                                     std=[0.229, 0.224, 0.225])
and get  Mean AP = 0.7217 
But forgot about BGR to RGB, I check it today

**Comment by jwyang** [comment]
Hi,  @dereyly, have you obtained the correct performance?

**Comment by dereyly** [comment]
@jwyang  With fix BGR->RGB i get slightly better result mAP=0.729

**Comment by jwyang** [comment]
ok, it's still far away from the caffe net. Did you normalize the image rob value to [0,1] before the normalization.

**Comment by dereyly** [comment]
@jwyang yes. I use it
```python
 padding_data = self.normalize(padding_data.div(255))
```
code details: https://github.com/dereyly/faster-rcnn.pytorch

**Comment by jwyang** [comment]
On your GitHub, it seems that the pixel_mean was not set to [0,0,0]? You did on your local, right? You also transpose the color channel as well, right? I will train the model on my side.

**Comment by dereyly** [comment]
I set mean_pixel [[0,0,0]](https://github.com/dereyly/faster-rcnn.pytorch/blob/c7b790d2bda76824448db87ffb2fe6684b943421/trainval_net.py#L208)
First, I transpose color, then I just comment [bgr](https://github.com/dereyly/faster-rcnn.pytorch/blob/c7b790d2bda76824448db87ffb2fe6684b943421/lib/roi_data_layer/minibatch.py#L73)

**Comment by jwyang** [comment]
@dereyly have you get better performance using pytorch version resnet?

**Comment by dereyly** [comment]
Hi @jwyang 
Max precision still mAP=0.729
Now i have no Idea how to eliminate this gap
Maybe decrease WD to 1e-4 (now is 5e-4, but its no big difference)

**Comment by jwyang** [comment]
I see, i will try on my side

**Comment by jwyang** [comment]
Hi, @dereyly , I tried the pytorch pretrained model on my side. It seems that pytorch model has a gap to caffe pre-trained model for sure. So just use caffe model,:)

**Comment by ZhangHanbo** [comment]
> Hi, @dereyly , I tried the pytorch pretrained model on my side. It seems that pytorch model has a gap to caffe pre-trained model for sure. So just use caffe model,:)

Hi, @jwyang, I have tried Pytorch pretrained ResNet101, and got a mAP of 72.57. Since I have modified the code, so I want to make sure whether it matches your result on Pytorch pretrained model?
